Upstream build against GTK 4.1.0; ours is just a touch later at > 4.10

--- a/ui/gtk/gsk.sigs
+++ b/ui/gtk/gsk.sigs
@@ -1,17 +1,17 @@
-GskRenderNodeType gsk_render_node_get_node_type(GskRenderNode* node);
+GskRenderNodeType gsk_render_node_get_node_type(const GskRenderNode* node);
 void gsk_render_node_unref(GskRenderNode* node);
-GskRenderNode* gsk_transform_node_get_child(GskRenderNode* node);
-GskRenderNode* gsk_opacity_node_get_child(GskRenderNode* node);
-GskRenderNode* gsk_color_matrix_node_get_child(GskRenderNode* node);
-GskRenderNode* gsk_repeat_node_get_child(GskRenderNode* node);
-GskRenderNode* gsk_clip_node_get_child(GskRenderNode* node);
-GskRenderNode* gsk_rounded_clip_node_get_child(GskRenderNode* node);
-GskRenderNode* gsk_shadow_node_get_child(GskRenderNode* node);
-GskRenderNode* gsk_blur_node_get_child(GskRenderNode* node);
-GskRenderNode* gsk_debug_node_get_child(GskRenderNode* node);
-GskRenderNode* gsk_container_node_get_child(GskRenderNode* node, guint idx);
-GskRenderNode* gsk_gl_shader_node_get_child(GskRenderNode* node, guint idx);
-guint gsk_container_node_get_n_children(GskRenderNode* node);
-guint gsk_gl_shader_node_get_n_children(GskRenderNode* node);
-GdkTexture* gsk_texture_node_get_texture(GskRenderNode* node);
+GskRenderNode* gsk_transform_node_get_child(const GskRenderNode* node);
+GskRenderNode* gsk_opacity_node_get_child(const GskRenderNode* node);
+GskRenderNode* gsk_color_matrix_node_get_child(const GskRenderNode* node);
+GskRenderNode* gsk_repeat_node_get_child(const GskRenderNode* node);
+GskRenderNode* gsk_clip_node_get_child(const GskRenderNode* node);
+GskRenderNode* gsk_rounded_clip_node_get_child(const GskRenderNode* node);
+GskRenderNode* gsk_shadow_node_get_child(const GskRenderNode* node);
+GskRenderNode* gsk_blur_node_get_child(const GskRenderNode* node);
+GskRenderNode* gsk_debug_node_get_child(const GskRenderNode* node);
+GskRenderNode* gsk_container_node_get_child(const GskRenderNode* node, guint idx);
+GskRenderNode* gsk_gl_shader_node_get_child(const GskRenderNode* node, guint idx);
+guint gsk_container_node_get_n_children(const GskRenderNode* node);
+guint gsk_gl_shader_node_get_n_children(const GskRenderNode* node);
+GdkTexture* gsk_texture_node_get_texture(const GskRenderNode* node);
 float gsk_opacity_node_get_opacity(const GskRenderNode* node);
--- a/ui/gtk/gtk_util.cc
+++ b/ui/gtk/gtk_util.cc
@@ -66,7 +66,7 @@ GskRenderNode* GetRenderNodeChild(GskRenderNode* node) {
 std::vector<GskRenderNode*> GetRenderNodeChildren(GskRenderNode* node) {
   std::vector<GskRenderNode*> result;
   size_t n_children = 0;
-  GskRenderNode* (*get_child)(GskRenderNode*, guint) = nullptr;
+  GskRenderNode* (*get_child)(const GskRenderNode*, guint) = nullptr;
   switch (gsk_render_node_get_node_type(node)) {
     case GSK_CONTAINER_NODE:
       n_children = gsk_container_node_get_n_children(node);
